src/server/src/main/java/io/cassandrareaper/management/http/HttpMetricsProxy.java

```diff
@@ -17,49 +17,211 @@
+    // Not mandatory for our first iteration.
+    return Collections.emptyList();
   }
 
   @Override
-  public List<GenericMetric> collectPercentRepairedMetrics(Node node, String keyspaceName)
+  public List<GenericMetric> collectPercentRepairedMetrics(String keyspaceName)
       throws JMException, IOException {
     throw new UnsupportedOperationException("Not implemented");
   }
+
+  private Map<String, List<JmxStat>> collectMetrics() throws IOException {
+    return collectMetric(Optional.empty());
+  }
+
+  private Map<String, List<JmxStat>> collectMetric(Optional<String> metricName) throws IOException {
+    String url = String.format("http://%s:%s/metrics", proxy.getHost(), proxy.getMetricsPort());
+    if (metricName.isPresent()) {
+      url += "?name=" + metricName.get();
+    }
+    Request request = new Request.Builder()
+        .url(url)
+        .build();
+    String metrics;
+    try (Response response = this.httpClient.newCall(request).execute()) {
+      if (!response.isSuccessful()) {
+        throw new IOException("Unexpected code " + response);
+      }
+      metrics = response.body().string();
+      return parsePrometheusMetrics(metrics);
+    }
+  }
+
+  public List<GenericMetric> collectTpPendingTasks() throws ReaperException {
+    // Collect metrics with the tpstats pending tasks name
+    List<GenericMetric> genericMetrics;
+    try {
+      genericMetrics
+          = MetricsProxy.convertToGenericMetrics(collectMetric(Optional.of(TPSTATS_PENDING_METRIC_NAME)), node);
+    } catch (IOException e) {
+      throw new ReaperException("Error collecting metrics", e);
+    }
+    return genericMetrics.stream()
+        .filter(metric -> metric.getMetricType().equals("ThreadPools"))
+        .collect(Collectors.toList());
+  }
+
+  @VisibleForTesting
+  public Map<String, List<JmxStat>> parsePrometheusMetrics(String metrics) {
+    Map<String, List<JmxStat>> parsedMetrics = Maps.newHashMap();
+
+    //Regex Pattern for Prometheus Metrics format
+    Pattern pattern = Pattern.compile("(\\w+)(\\{.*\\})?\\s(\\d+\\.\\d+)");
+    Matcher matcher = pattern.matcher(metrics);
+
+    //Pattern to extract labels
+    Pattern labelPattern = Pattern.compile("(\\w+)=\"(.*?)\"");
+
+    while (matcher.find()) {
+      String metricName = matcher.group(1);
+      double value = Double.parseDouble(matcher.group(3));
+
+      // check if there are any labels and parse them
+      Map<String, String> labels = Maps.newHashMap();
+      String labelString = matcher.group(2);
+      if (labelString != null) {
+        Matcher labelMatcher = labelPattern.matcher(labelString);
+        while (labelMatcher.find()) {
+          labels.put(labelMatcher.group(1), labelMatcher.group(2));
+        }
+      }
+
+      parsedMetrics.computeIfAbsent(metricName, k -> new ArrayList<JmxStat>());
+
+      Optional<JmxStat> metric = convertToJmxStat(metricName, labels, value);
```

Comment:
question: Why are these metrics converted to JmxStat instead of creating directly the GenericMetric class, which is what the methods will return? I feel like now we have to create a refactoring ticket to remove JmxStat from HttpMetrics as this class isn't doing any JMX work.

Reply:
Right, I was under the impression initially that the interface was exposing JmxStats as return type, but collectMetrics isn't defined in it and is a private method.
I'll get rid of JmxStat there.